Doc/reference/datamodel.rst

```diff
@@ -1962,6 +1960,53 @@ Notes on using *__slots__*
   of the iterator's values. However, the *__slots__* attribute will be an empty
   iterator.
 
+Per-attribute docstrings with *__slots__*
+"""""""""""""""""""""""""""""""""""""""""
+
+If a :class:`dictionary <dict>` is used to specify *__slots__*, the values of
+the dictionary can be used to specify docstrings for each attribute. For
+example::
+
+  class Card:
+      """A card from a standard French deck"""
+
+      __slots__ = {
+          'suit': 'Either "Spades", "Hearts", "Clubs" or "Diamonds"',
+          'rank': 'A positive integer in the range 2 <= x <= 14'
+      }
+
+      def __init__(self, suit, rank):
+          self.suit = suit
+          self.rank = rank
+
+Calling :func:`help` on this class will yield the following output::
+
+  >>> help(Card)
+  Help on class Card in module __main__:
+
+  class Card(builtins.object)
+   |  Card(suit, rank)
+   |
+   |  A card from a standard French deck
+   |
+   |  Methods defined here:
+   |
+   |  __init__(self, suit, rank)
+   |      Initialize self.  See help(type(self)) for accurate signature.
+   |
+   |  ----------------------------------------------------------------------
+   |  Data descriptors defined here:
+   |
+   |  rank
+   |      A positive integer in the range 2 <= x <= 14
+   |
+   |  suit
+   |      Either "Spades", "Hearts", "Clubs" or "Diamonds"
+
+
+.. versionadded:: 3.8
```

Comment:
One more note: it was not clear to me which feature was added in 3.8 on the first sight. Firstly, I thought about `__slots__ = {}` notation. Not about `help()` support.

I even had to test this to make sure mypy works correctly for this case 😆

Reply:
Should be fixed now 👍